DigitalLearningSolutions.Data/DataServices/UserDataService/AdminUserDataService.cs

```diff
@@ -272,7 +272,7 @@ public void DeactivateAdmin(int adminId)
 
         public void DeleteAdminAccount(int adminId)
         {
-            int existingId = (int)connection.ExecuteScalar("SELECT aa.UserID FROM AdminAccounts AS aa INNER JOIN SupervisorDelegates AS sd ON aa.ID = sd.SupervisorAdminID WHERE aa.ID=@adminId", new { adminId });
+            int? existingId = (int?)connection.ExecuteScalar("SELECT aa.UserID FROM AdminAccounts AS aa INNER JOIN SupervisorDelegates AS sd ON aa.ID = sd.SupervisorAdminID WHERE aa.ID=@adminId", new { adminId });
 
             if (existingId > 0)

```

Comment:
What happens here if existingId IS NULL?

Reply:
The null will only occur when there ae no records in SupervisorDelegates table which is associated with AdminAccounts table. In this scenario, the users table will not be updated and the only action executed will be to delete the record from AdminAccounts table. Now what we can do is that if existingId is null then get the userid from AdminAccounts table and make that user inactive (Active = 0). Please let me know if that approach makes sense

Reply:
@manishagarwalnhs, no change needed here - just getting clarity on how the null was handled - sounds sensible already.